### PR TITLE
python3Packages.snakemake: init at 6.15.2

### PR DESCRIPTION
**Pull request by wd15**
###### Motivation for this change

Add Snakemake workflow tool, see https://snakemake.readthedocs.io

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wd15** [comment]
This is failing to build as pythonPackages.google-cloud-storage is failing on master. Can't go ahead with this until that's fixed.

File: pkgs/development/python-modules/snakemake/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+, ratelimiter
+, smart-open
+, stopit
+, tabulate
+, toposort
+, wrapt
+, pandas
+, pytest
+, requests-mock
+}:
```
Comment (jonringer) [inline pkgs/development/python-modules/snakemake/default.nix:24]: NIT, sorry
```suggestion
}:

```

File: pkgs/development/python-modules/snakemake/default.nix [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+buildPythonPackage rec {
+  pname = "snakemake";
+  version = "6.15.2";
+
+  src = fetchFromGitHub {
+    owner = "snakemake";
+    repo = pname;
+    rev = "v" + version;
```
Comment (jonringer) [inline pkgs/development/python-modules/snakemake/default.nix:32]: most common way to express this:
```suggestion
    rev = "v${version}";
```

File: pkgs/development/python-modules/snakemake/default.nix [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  checkInputs = [
+    pandas
+    pytest
+    requests-mock
+  ];
+
+  checkPhase = ''
+    py.test --ignore tests/test_tes.py --ignore tests/test_tibanna.py
+  '';
```
Comment (jonringer) [inline pkgs/development/python-modules/snakemake/default.nix:64]:
```suggestion
  checkInputs = [
    pandas
    pytestCheckHook
    requests-mock
  ];

  disabledTestPaths = [
    "tests/test_tes.py"
    "tests/test_tibanna.py"
  ];
```

**Comment by veprbl** [comment]
https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/science/misc/snakemake/default.nix

File: pkgs/development/python-modules/snakemake/default.nix [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+, ratelimiter
+, smart-open
+, stopit
+, tabulate
+, toposort
+, wrapt
+, pandas
+, pytestCheckHook
+, requests-mock
+}:
```
Comment (jonringer) [inline pkgs/development/python-modules/snakemake/default.nix:24]: This is a big boy...

**Comment by jonringer** [comment]
looks like the already packaged application is already a fit. Maybe we can just update it?

**Comment by wd15** [comment]
> looks like the already packaged application is already a fit. Maybe we can just update it?

Wow, sorry. I completely missed that. Yup, will update it and see if we can switch on a few tests.

**Pull request closed without merging**